microservices/s3_to_redshift/asset_data_to_redshift.py

```diff
@@ -316,26 +339,46 @@ def is_processed(object_summary):
                 logger.exception("S3 transfer failed.\n{0}".format(e.message))
             continue
 
-    # Check for an empty file. If it's empty, accept it as good and skip
+    # Check for an empty file. If it's empty, accept it as bad and skip
     # to the next object to process
     try:
-        df = pd.read_csv(StringIO(csv_string), sep=delim, index_col=False,
-                         dtype=dtype_dic, usecols=range(column_count))
-    except Exception as e:
+        df = pd.read_csv(
+            StringIO(csv_string),
+            sep=delim,
+            index_col=False,
+            dtype=dtype_dic,
+            usecols=range(column_count))
+    except pandas.errors.EmptyDataError as _e:
         logger.exception('exception reading {0}'.format(object_summary.key))
-        if (str(e) == "No columns to parse from file"):
-            logger.warning('File is empty, keying to goodfile \
+        if (str(_e) == "No columns to parse from file"):
+            logger.warning('File is empty, keying to badfile \
                            and proceeding.')
-            outfile = goodfile
+            outfile = badfile
         else:
             logger.warning('File not empty, keying to badfile \
                            and proceeding.')
             outfile = badfile
-        client.copy_object(Bucket="sp-ca-bc-gov-131565110619-12-microservices",
-                           CopySource="sp-ca-bc-gov-\
-                           131565110619-12-microservices/"
-                           + object_summary.key, Key=outfile)
-        continue
```

Comment:
@BJenning does it make sense to halt the microservice operation if the outfile = badfile here and posts an exit code of 1 instead?

Reply:
Isn't that what lines 362 to 369 are doing? Copying the file to the bad folder, and then calling clean_exit...that should handle the pandas error?

Reply:
> Isn't that what lines 362 to 369 are doing? Copying the file to the bad folder, and then calling clean_exit...that should handle the pandas error?

You're correct, I misread the commit record when I made this comment.